routers/repo/view.go

```diff
@@ -330,7 +330,17 @@ func renderDirectory(ctx *context.Context, treeLink string) {
 				if markupType := markup.Type(readmeFile.name); markupType != "" {
 					ctx.Data["IsMarkup"] = true
 					ctx.Data["MarkupType"] = string(markupType)
-					ctx.Data["FileContent"] = string(markup.Render(readmeFile.name, buf, readmeTreelink, ctx.Repo.Repository.ComposeDocumentMetas()))
+					var result strings.Builder
+					err := markup.Render(&markup.RenderContext{
```

Comment:
It may be nice to have something like a fallback instead of a hard error [to support this](https://github.com/go-gitea/gitea/pull/14661#issuecomment-778622853)